RecoPixelVertexing/PixelTrackFitting/plugins/PixelTrackProducerFromSoA.cc

```diff
@@ -160,16 +160,23 @@ void PixelTrackProducerFromSoA::produce(edm::StreamID streamID,
 
   int32_t nt = 0;
 
-  for (int32_t it = 0; it < nTracks; ++it) {
+  //sort index by pt
+  std::vector<int32_t> sortIdxs(nTracks);
+  std::iota(sortIdxs.begin(), sortIdxs.end(), 0);
+  std::sort(
+      sortIdxs.begin(), sortIdxs.end(), [&](size_t const i1, size_t const i2) { return tsoa.pt(i1) > tsoa.pt(i2); });
```

Comment:
```suggestion
      sortIdxs.begin(), sortIdxs.end(), [&](int32_t const i1, int32_t const i2) { return tsoa.pt(i1) > tsoa.pt(i2); });
```
Minor comment/question (not necessarily a suggestion):
  - iiuc, the `pt(i)` call uses `int32_t` (not `size_t`) from [`eigenSoA`](https://github.com/cms-sw/cmssw/blob/6d2f66057131baacc2fcbdd203588c41c885b42c/HeterogeneousCore/CUDAUtilities/interface/eigenSoA.h#L21).
  - also, `nt` above is `int32_t`, but it is used below to fill entries of `indToEdm`, which stores [`uint16_t`](https://github.com/cms-sw/cmssw/blob/76a217d1d9f68b71353de4e065d75d0361a0249d/RecoPixelVertexing/PixelTrackFitting/plugins/PixelTrackProducerFromSoA.cc#L42).

I image none of this has any practical implication.